designs/cloudWatch/cloudWatchIntegration.md

```diff
@@ -0,0 +1,77 @@
+# CloudWatch Logs Experience
+
+* **Type**: UX Design
+* **Status**: Proposal
+
+## Introduction
+CloudWatch Logs enables a user to centralize the logs from all of their systems, applications, and AWS services that they use, in a single, highly scalable service. This document describes adding CloudWatch Logs functionality into the JetBrains toolkit in order to facilitate easily viewing and searching them for specific error codes or patterns.
+
+## Comparison to other products
+
+### AWS Toolkit for VSCode  
+Not supported
+
+### AWS Toolkit for Visual Studio  
+Supports listing of Lambda CloudWatch Log Streams and downloading them to a temporary file and opening it in Notepad
+
+## Experience
+
+### View Log Groups
+The AWS Explorer gains a CloudWatch root node for interacting wit CloudWatch service. The node lists all the Log Groups that exist in the selected region for the selected credential profile as child nodes. Double clicking a Log Group opens the CloudWatch tool window with the tab for the Log Streams that belong to that group. A new tab is in the tool window is created if one does not already exist.
```

Comment:
Log group names can be very very long. What's the strategy for displaying them? (truncate, group by prefix, etc.)

Reply:
There isn't any harm in long names in the tree, they get truncated to the width of the tool window, and show the full value when moused over. Speedsearch also still searches the "hidden" parts

Reply:
My fear is that information that shows up (without hover) might just be a long prefix.

e.g. the common paradigm is `/aws/<service-name>`

If the service name is pretty long (like `/aws/kinesisfirehose/`) the user might just see a bunch of `/aws/kinesisfir..` and wouldn't be able to discern which one to choose at a glance.

Reply:
should we split on common separators and build a hierarchy? (e.g `/`) similar to s3? if we did that I think that takes us back to the toolwindow for finding a log-group. The names are max 512 characters (which I agree is pretty long) but I think for most use-cases it'll be OK.